docker-compose.yaml

```diff
@@ -43,20 +43,27 @@ services:
       - local
 
   ray:
-    image: rayproject/ray:2.30.0-py311-cpu${RAY_ARCH_SUFFIX}
+    image: rayproject/ray:2.30.0-py311${COMPUTE_TYPE}${RAY_ARCH_SUFFIX}
     ports:
       - "6379:6379"
       - "8265:8265"
       - "10001:10001"
     # https://docs.ray.io/en/releases-2.30.0/cluster/cli.html#ray-start for more info about the command
     command: bash -c "ray start --head --dashboard-port=8265 --port=6379 --dashboard-host=0.0.0.0 --ray-client-server-port 10001 --block" # pragma: allowlist secret
     shm_size: 2g
+    runtime: nvidia
     deploy:
       resources:
         limits:
           cpus: '2'
           memory: '5g'
+        reservations:
+          devices:
+            - driver: nvidia
+              count: 1
+              capabilities: [gpu]
     environment:
+      - NVIDIA_VISIBLE_DEVICES=all
```

Comment:
I'd suggest making `all` and device count dynamic, such as `${GPU_COUNT:-0}` and `NVIDIA_VISIBLE_DEVICES=${NVIDIA_VISIBLE_DEVICES:-all}`.